decidim-admin/app/commands/decidim/admin/reorder_participatory_process_steps.rb

```diff
@@ -0,0 +1,54 @@
+# frozen_string_literal: true
+module Decidim
+  module Admin
+    # A command that sets a step in a participatory process as active (and
+    # unsets a previous active step)
+    class ReorderParticipatoryProcessSteps < Rectify::Command
+      # Public: Initializes the command.
+      #
+      # collection - an ActiveRecord::Relation of steps
+      # order_string - a String representing an Array holding the order of IDs
+      #   of steps
+      def initialize(collection, order_string)
+        @collection = collection
+        @order_string = order_string
+      end
+
+      # Executes the command. Braodcasts these events:
+      #
+      # - :ok when everything is valid.
+      # - :invalid if the data wasn't valid and we couldn't proceed.
+      #
+      # Returns nothing.
+      def call
+        return broadcast(:invalid) unless order_string.present?
+        begin
+          @order = JSON.parse(order_string)
```

Comment:
I'd move this to a method (`order`). This way you can do `return broadcast(:invalid) unless order` and encapsulate all the JSON and presence logic in a method.